### PR TITLE
[PLAYER-4272] added sample in AdvancedPlaybackSampleApp for proper fa…

### PR DESCRIPTION
**Pull request by SergeyBicarte**
This sample demonstrate proper  setEmbedCode() method  using for fast channel switching (fast tapping on button in activity).

### CODE REVIEW

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Observable;
+import java.util.Observer;
+
+/**
+ * This activity illustrates how you can create, play a basic playback video
```
Comment (pilievOoyala) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java]: Small change: 
This activity illustrates how you can create a player, play a basic video
and how to quickly switch between streams by using the setEmbedCode() method.

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+        final FCCTVRatingConfiguration tvRatingConfiguration = new FCCTVRatingConfiguration.Builder().setDurationSeconds(5).build();
+
+        player = new OoyalaPlayer(PCODE, domain, this, new Options.Builder().setTVRatingConfiguration( tvRatingConfiguration ).setBypassPCodeMatching(true).setUseExoPlayer(true).build());
+        OptimizedOoyalaPlayerLayoutController layoutController = new OptimizedOoyalaPlayerLayoutController(
+                (OoyalaPlayerLayout) findViewById(R.id.player),
+                player,
+                DefaultControlStyle.AUTO
+        );
+        player.setAdsSeekable(true); // this will help us skip ads if need be.
```
Comment (pilievOoyala) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java]: Isn't this also going to let you manually skip around during an ad?

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        player.addObserver(this);
+        player.setVolume(.05f);
+        int r = player.beginFetchingAdvertisingId(this, this);
+        Log.d( TAG, "initAdvertisingId: " + r );
+    }
+
+    public void onAdvertisingIdSuccess( String adId ) {
```
Comment (pilievOoyala) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java]: Do we write out the whole word 'Advertising' anywhere else? This doesn't look like an override so you can shorten it to on AdIdSuccess. Same with the onAdvertisingIdError method below.

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        } else if (notificationName == OoyalaPlayer.TIME_CHANGED_NOTIFICATION_NAME) {
+        }
+    }
+
+    @Override
+    public void getTokenForEmbedCodes(List<String> embedCodes,
+                                      EmbedTokenGeneratorCallback callback) {
+        String embedCodesString = "";
+        for (String ec : embedCodes) {
+            if(ec.equals("")) embedCodesString += ",";
```
Comment (pilievOoyala) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java]: please separate this out onto two lines and use braces.

**Comment by brol1dev** [comment]
Holding this one until we release. Shouldn't be part of 4.39.

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java (hunk 1 above)
Comment (SergeyBicarte) [reply AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java]: done

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java (hunk 2 above)
Comment (SergeyBicarte) [reply AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java]: This sample was cloned from SDK ooyalaTestApp.  We don't  need here ads , so i removed all pieces of ads code.

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java (hunk 3 above)
Comment (SergeyBicarte) [reply AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java]: This sample was cloned from SDK ooyalaTestApp.  We don't  need here ads , so i removed all pieces of ads code.

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java (hunk 4 above)
Comment (SergeyBicarte) [reply AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java]: done

**Review by akshaydch11 — APPROVED** [review]
LGTM

**Review by brol1dev — CHANGES_REQUESTED** [review]
I don't understand this PR. @SergeyBicarte I think the additional code is catching a thread exception which means we still get errors when fast changing embed codes? Not sure what we're trying to show to a user here. Can you explain further? If you want you can create a meeting.

**Comment by SergeyBicarte** [comment]
This sample is answer for PLAYER-4272 request "Could you provide information how to correct switch channels or fix this bug?". It  demonstrate how you can fast switching channels by  setEmbedCode() method and do not crash.

**Comment by DmitriiMaslov** [comment]
@rcvrgs @akshaydch11 Could you please review Sergey's comment, We need to understand what is going on with this PR. It was supposed to get in 4.40 and we are in this release now.

**Pull request merged**